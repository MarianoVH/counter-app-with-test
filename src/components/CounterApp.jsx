import React, { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const addValue = () => setCounter(counter + 1);
  const subtractValue = () => setCounter(counter - 1);
  const reset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={addValue}>+1</button>
      <button onClick={reset}>reset</button>
      <button onClick={subtractValue}>-1</button>
    </>
  );
};

CounterApp.propType = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 0,
};
