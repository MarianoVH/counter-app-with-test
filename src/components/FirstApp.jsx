import React from "react";
import PropTypes from "prop-types";

export const FirstApp = ({ subTitle, title }) => (
  <>
    <h1>{title}</h1>
    <h2>{subTitle}</h2>
  </>
);

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  subTitle: "I'm a subtitle",
};
