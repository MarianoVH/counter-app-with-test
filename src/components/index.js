export * from "./CounterApp";
export * from "./FirstApp";
