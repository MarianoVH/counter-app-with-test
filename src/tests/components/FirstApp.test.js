import React from "react";
import { shallow } from "enzyme";

import { FirstApp } from "../../components";

describe("Test on <FirstApp />", () => {
  test("It should render <FirstApp /> ", () => {
    const title = "This is the title";
    const wrapper = shallow(<FirstApp title={title} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("It should render <FirstApp /> ", () => {
    const title = "This is the title";
    const subTitle = "This is the subTitle";
    const wrapper = shallow(<FirstApp title={title} subTitle={subTitle} />);

    const textInP = wrapper.find("h2").text();

    expect(textInP).toBe(subTitle);
  });
});
