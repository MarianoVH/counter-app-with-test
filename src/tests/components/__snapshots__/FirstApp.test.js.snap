// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <FirstApp /> It should render <FirstApp />  1`] = `
<Fragment>
  <h1>
    This is the title
  </h1>
  <h2>
    I'm a subtitle
  </h2>
</Fragment>
`;
